pkg/modules/authorization.go

```diff
@@ -1218,19 +1237,26 @@ func applyDeleteAuthorizationProxyServerV2(ctx context.Context, isDeleting bool,
 					redisPasswordKey = config.RedisPasswordKey
 				}
 			}
+		case AuthConfigSecretComponent:
+			// create config signing kubernetes secret or use a secret provider class
+			for _, config := range component.ConfigSecretProviderClass {
+				configSecretProviderClassName = config.ConfigSecretProviderClassName
+			}
 		default:
 			continue
 		}
 	}
 
 	// conversion to int32 is safe for a value up to 2147483647
 	// #nosec G115
-	deployment := getProxyServerScaffold(cr.Name, sentinelName, cr.Namespace, proxyImage, opaImage, opaKubeMgmtImage, redisSecretName, redisPasswordKey, int32(replicas), redisReplicas)
+	deployment := getProxyServerScaffold(cr.Name, sentinelName, cr.Namespace, proxyImage, opaImage, opaKubeMgmtImage, configSecretProviderClassName, redisSecretName, redisPasswordKey, int32(replicas), redisReplicas)
 
 	if redisSecretProviderClassName != "" && redisSecretName != "" {
 		mountRedisVolumes(&deployment.Spec.Template.Spec, redisSecretProviderClassName)
```

Comment:
Not due to your changes. If you could rename this to _mountRedisVolume_(), it would be great.

Reply:
Changed to mountSPCVolume()